leo/unittests/test_writers.py

```diff
@@ -50,6 +52,48 @@ def test_dart_writer(self):
         x = DartWriter(c)
         x.write(root)
     #@-others
+#@+node:ekr.20231219151314.1: ** class TestMDWriter(BaseTestWriter)
+class TestMarkdownWriter(BaseTestWriter):
+    """Test Cases for the markdown writer plugin."""
+    #@+others
+    #@+node:ekr.20231219151402.1: *3* TestMDWriter.test_markdown_writer
+    def test_markdown_writer(self):
+        
+        ### Should be a round-trip test.
+        
+        g.trace('=====')
+
+        contents = textwrap.dedent("""
+            # 1st level title X
+
+            some text in body X
+
+            ## 2nd level title Z
+
+            some text in body Z
+
+            # 1st level title A
+
+            ## 2nd level title B 
+
+            some body content of the 2nd node 
+        """).strip() + '\n'
+
+        c, root = self.c, self.c.p
+        child = root.insertAsLastChild()
+        child.h = 'h'
+        child.b = contents
```

Comment:
The contents is not to be put into a body as a whole directly, it's loaded from an @auto node, not an @edit node. So it builds a tree when 'refreshed from file'. (Could it be that perhaps the extra newline can be omitted when reading into the tree instead of when writing as I first proposed?)

Reply:
I agree with your analysis.  I thought I had put a comment to exactly that effect here, but apparently I forgot. I'll get the test to fail soon :-)